controllers/account.js

```diff
@@ -57,15 +57,13 @@ module.exports.updatePassword = async (req, res) => {
           req.body.newPassword,
           (err) => {
             if (err) {
-              //error message
-              res.redirect("/account/change-password");
+              res.redirect("/account/change-password?feedback=incorrectPw");
```

Comment:
I'm not sure I like this approach tbh, you could probably send the error message straight to the template, no?

Reply:
Honestly I'm not too sure how i could otherwise pass the data together with the redirect and accessing it while rendering the page. Might need a bit more of an elaborate suggestions on how i could better change this.

Reply:
Passing some context with a redirect might be tricky indeed. However I was thinking now that you could probably handle the change password logic without causing additional redirecting. Take this idea as an example:

Client submits form, but instead of letting HTML default form submission event do it's thing, we could attach an event listener that listens whenever the user submits that form and executes our own custom form submit function (https://www.freecodecamp.org/news/how-to-submit-a-form-with-javascript/). In that function we would send a POST request to our backend, where the actual handling of changing the password is and send either a success or fail code back. When the client finishes waiting for the response of the change password request, a DOM manipulation event is executed, depending on the response (hide or show error messages, anything goes).

This is a very rough draft what it could look like:

```
function submitHandler() {
    fetch('<URL>', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        // your data here
      })
    })
    .then(response => response.json())
    .then(data => {
      document.querySelector('#your-element').style.display = 'block';
    })
    .catch(error => {
      console.error('Error:', error);
    });
}
```